WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/EditorPage.java

```diff
@@ -5,52 +5,99 @@
+    private static ViewInteraction addMediaButton = onView(withId(R.id.media_button_container));
+    private static ViewInteraction mediaLibraryButton = onView(withId(R.id.media_bar_button_library));
+    private static ViewInteraction allowMediaAccessButton = onView(allOf(withId(R.id.button),
+            withText("Allow")));
+    private static ViewInteraction confirmButton = onView(withId(R.id.mnu_confirm_selection));
 
     public EditorPage() {
         editor.check(matches(isDisplayed()));
     }
 
-    public EditorPage enterTitle(String postTitle) {
+    public void enterTitle(String postTitle) {
         titleField.perform(typeText(postTitle), ViewActions.closeSoftKeyboard());
-
-        return this;
     }
 
-    public EditorPage enterContent(String postContent) {
+    public void enterContent(String postContent) {
         editor.perform(typeText(postContent), ViewActions.closeSoftKeyboard());
+    }
+
+    // Image needs a little time to be uploaded after entering the image
+    public void enterImage() {
+        // Click on add media button
+        clickOn(addMediaButton);
+        clickOn(mediaLibraryButton);
+
+        if (isElementDisplayed(allowMediaAccessButton)) {
+            // Click on Allow button
+            clickOn(allowMediaAccessButton);
+        }
+
+        // Click on a random image
+        onView(withIndex(withId(R.id.media_grid_item_image), 0)).perform(click());
+
+        // Click the confirm button
+        clickOn(confirmButton);
+
+        if (isElementDisplayed(onView(withText("LEAVE OFF")))) {
+            // Accept alert for media access
+            clickOn(onView(withText("LEAVE OFF")).inRoot(isDialog()));
+        }
+    }
+
+    public void openSettings() {
+        openActionBarOverflowOrOptionsMenu(getInstrumentation().getTargetContext());
+        clickOn(onView(withText("Post settings")));
+    }
+
+    public void addACategory(String category) {
+        clickOn(onView(withId(R.id.post_categories_container)));
+        clickOn(onView(withText(category)));
+        pressBack();
+    }
+
+    public void addATag(String tag) {
+        clickOn(onView(withId(R.id.post_tags_container)));
+        ViewInteraction tagsField = onView(withId(R.id.tags_edit_text));
+        populateTextField(tagsField, tag);
+        pressBack();
+    }
 
-        return this;
+    public void setFeaturedImage() {
+        clickOn(onView(withId(R.id.post_add_featured_image_button)));
+        clickOn(onView(withId(R.id.icon_wpmedia)));
+        onView(withIndex(withId(R.id.media_grid_item_image), 0)).perform(click());
     }
 
     public boolean publishPost() {
-        publishButton.perform(click());
-        onView(withText("PUBLISH NOW"))
-                .inRoot(isDialog())
-                .check(matches(isDisplayed()))
-                .perform(click());
+        clickOn(publishButton);
+        clickOn(onView(withText("PUBLISH NOW")));
         waitForElementToBeDisplayed(publishConfirmation);
```

Comment:
I'm seeing intermittent failures here. It seems that there is a race condition that causes the publishing logic not to work properly if the network request takes too long. So the mocks are too fast 😅

I'll make a suggested workaround for this below.

Reply:
😄 it knows you're trying to trick it into thinking it's talking to a real API

Reply:
I've committed in your below suggestion.